### PR TITLE
JBPM-5242 and JBPM-5243 - Case overview and details

### PR DESCRIPTION
**Pull request by cristianonicolai**
New screens to show case instance related data.

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@krisv @nmirasch @mswiderski another set of screens around case instance. Case overview is the placeholder for all other UI building blocks, starting with case details.

File: jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-api/src/main/java/org/jbpm/console/ng/cm/model/CaseInstanceSummary.java [hunk 1]
```diff
@@ -68,16 +73,55 @@ public void setContainerId(String containerId) {
...
 
+    public String getStatusString(){
+        if(status == null){
+            return "";
+        }
+        switch (status){
+            case 1:
+                return "Active";
+            default:
+                return "";
```
Comment (mswiderski) [inline jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-api/src/main/java/org/jbpm/console/ng/cm/model/CaseInstanceSummary.java]: Don't we need any other statuses?

File: jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/overview/CaseOverviewPresenter.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    @OnOpen
+    public void onOpen() {
+        view.addCaseDetails(CaseDetailsPresenter.SCREEN_ID, place.getParameters());
+//        view.addCaseActions(CaseActionsPresenter.SCREEN_ID, place.getParameters());
+//        view.addCaseStages(CaseStagesPresenter.SCREEN_ID, place.getParameters());
+//        view.addCaseComments(CaseCommentsPresenter.SCREEN_ID, place.getParameters());
+//        view.addCaseFiles(CaseFilesPresenter.SCREEN_ID, place.getParameters());
+//        view.addCaseRoles(CaseRolesPresenter.SCREEN_ID, place.getParameters());
+//        view.addCaseMilestones(CaseMilestonesPresenter.SCREEN_ID, place.getParameters());
+//        view.addCaseActivities(CaseActivitiesPresenter.SCREEN_ID, place.getParameters());
```
Comment (mswiderski) [inline jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/overview/CaseOverviewPresenter.java]: Might be removed I guess, or?

**Comment by mswiderski** [comment]
@cristianonocolai I added some minor comments around backend as I can't really review properly client side ;)

Looks good in general, nice job!

**Comment by cristianonicolai** [comment]
@mswiderski thanks for checking... I've fixed both comments.

**Comment by nmirasch** [comment]
+1, Thanks!

File: jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/overview/CaseOverviewViewImpl.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+import org.uberfire.client.mvp.PlaceManager;
+import org.uberfire.mvp.impl.DefaultPlaceRequest;
+
+@Dependent
+@Templated(stylesheet = "CaseOverviewViewImpl.css")
+public class CaseOverviewViewImpl extends Composite implements CaseOverviewPresenter.CaseOverviewView {
+
+    @Inject
+    @DataField("case-details-content")
+    FlowPanel caseDetails;
```
Comment (csadilek) [inline jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/overview/CaseOverviewViewImpl.java]: @cristianonicolai we really need to stop using GWT widgets so much. Can you take a look at the various DOM wrappers we have: https://github.com/errai/errai/tree/master/errai-common/src/main/java/org/jboss/errai/common/client/dom

In this case a Div etc. would probably be enough.

**Comment by csadilek** [comment]
@cristianonicolai I will merge this, but please see my comment about GWT widget replacements.  I think it can be done in a separate refactoring/PR as likely other new functionality already merged is also affected. Thanks.

File: jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/overview/CaseOverviewViewImpl.java (hunk 3 above)
Comment (cristianonicolai) [inline jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/overview/CaseOverviewViewImpl.java]: @csadilek ok, will take a look on that for the case management bits...

**Pull request merged**